File: apple/Sources/ReactNativeWatchOS/Modifiers/PickerStyle.swift
import SwiftUI

/// SwiftUI `.pickerStyle(_:)`. Maps the `style` string to a picker style.
/// Unknown values fall back to `.automatic`.
enum RNWPickerStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("pickerStyle") { view, params, _ in
            switch params.string("style") {
            case "navigationLink":
                return AnyView(view.pickerStyle(.navigationLink))
            case "wheel":
                return AnyView(view.pickerStyle(.wheel))
            case "inline":
                return AnyView(view.pickerStyle(.inline))
            default:
                return AnyView(view.pickerStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/pickerStyle.ts
import { createModifier } from './createModifier';

export type PickerStyle = 'automatic' | 'navigationLink' | 'wheel' | 'inline';

/// SwiftUI `.pickerStyle(_:)`. Sets the visual style applied to `Picker`s
/// within the view. `'automatic'` defers to the platform default.
export function pickerStyle(style: PickerStyle = 'automatic') {
  return createModifier('pickerStyle', { style });
}
